### PR TITLE
Clean up of fields; have floating div render entire text box during edit

### PR DESCRIPTION
**Pull request by tmickel**
-Removes borderRect_ altogether as we're not using this.
-Fixes up positioning of fields. Instead of borderRect_, the click target (field path/shadow block) is now the "master copy" of the field position and size.
-Move some logic around border radii and colors for the fields so that we can access these properties when drawing the CSS-styled box.
-Finally, in field_textinput, implement the drawing, sizing, and styling for the CSS-styled floating div box.

The goal for this PR is to allow us to do animations on the box once it's focused. It makes it easier for us to customize and style text fields being edited.

I know the positioning of the box has a lot of magic numbers - as it did before. I'm pretty convinced it's as good as it's going to get, and I don't have great justifications for the numbers except for fiddling in each browser.

### CODE REVIEW

File: core/css.js [hunk 1]
```diff
@@ -288,14 +295,13 @@ Blockly.Css.CONTENT = [
 
   '.blocklyHtmlInput {',
     'border: none;',
-    'border-radius: 4px;',
     'font-family: sans-serif;',
     'height: 100%;',
     'margin: 0;',
     'outline: none;',
-    'padding: 0 1px;',
+    'padding: 2px 0px;',
```
Comment (rachel-fenichel) [inline core/css.js]: apparently google style guide says just use 0, not 0px

File: core/field.js [hunk 2]
```diff
@@ -314,10 +304,9 @@ Blockly.Field.prototype.getSize = function() {
  * @private
  */
 Blockly.Field.prototype.getScaledBBox_ = function() {
-  var bBox = this.borderRect_.getBBox();
-  // Create new object, as getBBox can return an uneditable SVGRect in IE.
```
Comment (rachel-fenichel) [inline core/field.js:318]: keep the comment.

File: core/field_colour.js [hunk 3]
```diff
@@ -70,7 +70,12 @@ Blockly.FieldColour.prototype.columns_ = 0;
  */
 Blockly.FieldColour.prototype.init = function(block) {
   Blockly.FieldColour.superClass_.init.call(this, block);
-  this.borderRect_.style['fillOpacity'] = 1;
+  // XXX: borderRect_ has been removed from the field.
```
Comment (rachel-fenichel) [inline core/field_colour.js]: issue number?

File: core/field_textinput.js [hunk 4]
```diff
@@ -131,7 +131,7 @@ Blockly.FieldTextInput.prototype.showEditor_ = function(opt_quietInput) {
   var htmlInput = goog.dom.createDom('input', 'blocklyHtmlInput');
   htmlInput.setAttribute('spellcheck', this.spellcheck_);
   var fontSize =
-      (Blockly.FieldTextInput.FONTSIZE * this.workspace_.scale) + 'pt';
+      (Blockly.FieldTextInput.FONTSIZE) + 'pt';
```
Comment (rachel-fenichel) [inline core/field_textinput.js:134]: Regardless of scale?

Comment (tmickel) [inline core/field_textinput.js:134]: The font size is effected now by the scaling of the entire div (it's scaled by workspace scale).

**Pull request merged**